engine/src/main/scala/cromwell/webservice/routes/MetadataRouteSupport.scala

```diff
@@ -153,13 +166,25 @@ object MetadataRouteSupport {
                                  (implicit timeout: Timeout,
                                   ec: ExecutionContext): Future[MetadataJsonResponse] = {
 
+    def recordHistoricalMetadataLookupLagMetric(endTime: Option[OffsetDateTime]): Unit = {
+      val timeSinceEndTime = endTime match {
+        case Some(et) => FiniteDuration(JDuration.between(et, OffsetDateTime.now()).toMillis, TimeUnit.MILLISECONDS)
+        case None => 0.seconds
+      }
+
+      val instrumentationPath = MetadataServiceActor.MetadataInstrumentationPrefix :+ "archiver" :+ "historical_metadata_lookup_lag"
+      val message = InstrumentationServiceMessage(CromwellTiming(CromwellBucket(ServicesPrefix.toList, instrumentationPath), timeSinceEndTime))
+      serviceRegistryActor ! message
+    }
```

Comment:
This is 100% personal preference and optional but I honestly prefer having helper methods like this be a private method further down instead of occupying line & cognitive space at the top of the high-level method

Reply:
I would usually defer to reviewers' preference on things like this, but I'm hesitating because there's already an existing method at the same level as this one (namely, `checkIfMetadataDeletedAndRespond`) which relies on a lot of shared context with the higher-level method so extracting is likely to generate a lot of passthrough-variables as side-effects.